dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/manager/DefaultExecutorRepository.java

```diff
@@ -140,9 +141,21 @@ private ExecutorService createExecutor(URL url) {
         return (ExecutorService) extensionAccessor.getExtensionLoader(ThreadPool.class).getAdaptiveExtension().getExecutor(url);
     }
 
+    private Map<Integer, ExecutorService> getExecutors(URL url) {
+        Map<Integer, ExecutorService> executors;
+        // if it's on the provider side and has user-defined services, use biz service executor first, avoid using internal executor
+        if (PROVIDER_SIDE.equalsIgnoreCase(url.getParameter(SIDE_KEY))
+            && data.containsKey(EXECUTOR_SERVICE_COMPONENT_KEY)) {
+            executors = data.get(EXECUTOR_SERVICE_COMPONENT_KEY);
+        } else {
+            executors = data.get(getExecutorKey(url));
+        }
+        return executors;
+    }
+
```

Comment:
I have a question that why `ServiceDescriptor serviceDescriptor = applicationModel.getInternalModule().getServiceRepository().lookupService(url.getServiceInterface());` not work?

Reply:
Because `url.getServiceInterface()` always returns MetadataService, when the MetadataService is first loaded.

Reply:
![image](https://user-images.githubusercontent.com/38170721/220063942-df1905f7-5e0a-4fb9-b737-f577c5a2d647.png)

The `PortUnificationExchanger#reset` method will return a new url instead of updating, so the Map in `AbstractPortUnificationServer#supportedUrls` will not be changed.

![image](https://user-images.githubusercontent.com/38170721/220063744-cfd5a94a-3550-4f0c-accb-3d7cbc1293dc.png)

in  `NettyPortUnificationServerHandler#decode` , **localUrl** is set by the value in **urlMapper**, not the updated **url**

![image](https://user-images.githubusercontent.com/38170721/220062213-c6948393-9f43-40f0-bccf-d795aee49ca9.png)

So the url is still MetadataService.